### PR TITLE
Throw error if query element doesn't end with END_OBJECT

### PR DESCRIPTION
**Pull request by javanna**
Followup to #20515 where we added validation that after we parse a query within a query element, we should not get a field name. Truth is that the only token allowed at that point is END_OBJECT, as our DSL allows only one single query within the query object:

```
{
  "query" : {
    "term" : { "field" : "value" }
  }
}
```

We can then check that after parsing of the query we have an end_object that closes the query itself (which we already do). Following that we can check that the query object is immediately closed, as there are no other tokens that can be present in that position.

See https://github.com/elastic/elasticsearch/pull/19791#discussion-diff-73647510 too

Relates to #20515

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTests.java [hunk 1]
```diff
@@ -370,7 +370,7 @@ public void testTooManyQueriesInObject() throws IOException {
         String query = "{\"bool\" : {\"" + clauseType
                 + "\" : { \"match\" : { \"foo\" : \"bar\" } , \"match\" : { \"baz\" : \"buzz\" } } } }";
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTests.java]: Could you indent this object so it is easier to read?

File: core/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java [hunk 2]
```diff
@@ -436,11 +437,10 @@ public void testInvalid() throws Exception {
                 "    }\n" +
                 "  } }";
         try (XContentParser parser = XContentFactory.xContent(restContent).createParser(restContent)) {
-            SearchSourceBuilder.fromXContent(createParseContext(parser),
-                    searchRequestParsers.aggParsers, searchRequestParsers.suggesters, searchRequestParsers.searchExtParsers);
-            fail("invalid query syntax multiple keys under query");
-        } catch (ParsingException e) {
-            assertThat(e.getMessage(), containsString("filters"));
+            ParsingException e = expectThrows(ParsingException.class, () -> SearchSourceBuilder.fromXContent
+                    (createParseContext(parser),
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java]: Having the `(` on this line is pretty strange. Can you move it back up?

**Review by nik9000 — APPROVED** [review]
Makes sense to me.

**Review by areek — APPROVED** [review]
LGTM! Maybe in the future we should just use `ObjectParser` for parsing the source to avoid these complex parsing logic

File: core/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java (hunk 2 above)
Comment (javanna) [inline core/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java]: yea I blame intellij ;)

File: core/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTests.java (hunk 1 above)
Comment (javanna) [inline core/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTests.java]: ok ok

**Pull request merged**